src/hardware/driver/adc_internal.rs

```diff
@@ -44,9 +44,13 @@ impl AdcInternal {
 
         let mut adc1 = adc1.enable();
         adc1.set_resolution(adc::Resolution::SIXTEENBIT);
+        // use reasonable sampling time that gives accurate results with headroom bat is also not too slow
```

Comment:
```suggestion
        // use reasonable sampling time that gives accurate results with headroom but is also not too slow
```

I'd add the actual calculation.
Also please start a branch where you port/cherry-pick changes like this one (which applies to Pounder CPU ADC channels as well) to master. Or at least note (file issue) that this should be done.